libknet/compress.c

```diff
@@ -209,6 +210,32 @@ static int compress_lib_test(knet_handle_t knet_h)
 		log_err(knet_h, KNET_SUB_COMPRESS, "Unable to compress test buffer. Please check your compression settings: %s", strerror(savederrno));
 		errno = savederrno;
 		return -1;
+	} else if ((long unsigned int)dst_comp_len >= KNET_DATABUFSIZE) {
+		/*
+		 * compress not effective, try again using default compression level when available
+		 */
+		request_level = knet_h->compress_level;
+		log_warn(knet_h, KNET_SUB_COMPRESS, "Requested compression level (%d) did not generate any compressed data \
+				(source: %zu destination: %zu)", request_level, sizeof(src), dst_comp_len);
+
+		if ((!compress_modules_cmds[knet_h->compress_model].ops->get_default_level()) ||
+				(knet_h->compress_level = compress_modules_cmds[knet_h->compress_model].ops->get_default_level() < 0)) {
```

Comment:
Instead of using < 0 check if it's == KNET_COMPRESS_UNKNOWN_DEFAULT so we are free to change values around if we need.